BDArmory/Ammo/PooledRocket.cs

```diff
@@ -776,7 +776,7 @@ void Detonate(Vector3 pos, bool missed, Part hitPart = null, bool airDetonation
                         {
                             for (int f = 0; f < 20; f++) //throw 20 random raytraces out in a sphere and see what gets tagged
```

Comment:
I think this would be more efficient with a `RaycastCommand` with 20 rays instead of doing them individually.
There's a commented out "Test RaycastCommand" test in BDArmorySetup.cs where I tested this on my machine and found the break-even point to be around 8 raycasts. It's likely different on different hardware though.

Shouldn't `rb.velocity` be `currentVelocity`? Also, there's no need to normalise it as the constructor for `Ray` applies normalisation anyway.

Reply:
That would make more sense. Why it was originally rb, no idea; this code is years old.

Looks to be about 10-ish raycasts is the breakpoint for me, but it would fluctuate by a couple of microsec depending on the exact background load on the system at that particular moment